src/productList/ProductList.jsx

```diff
@@ -0,0 +1,157 @@
+import React, { useState, useEffect, useTransition } from 'react'
+import { Card, Input, Spin, Alert, AutoComplete } from 'antd'
+import { reqProductList, reqSearchProduct } from '../api'
+import './produc-list.less'
+const { Meta } = Card;
+const { Search } = Input;
+
+let cutArr = [] //用来存放切割后的数组
+let cutArrKey = 1
+
+export default function ProductList() {
+    const [productList, setProductList] = useState([])
+    const [isLoading, setIsLoading] = useState(true)
+    const [option, setOption] = useState([])
+    const [open, setOpen] = useState(false)
+    const [isPending, starTransition] = useTransition()
+
+    useEffect(() => {
+        const getProductList = async () => {
+            const result = await reqProductList()
+            const { products } = result.data
+            handleCutArr(products)
+        }
+
+        //如果是first 才调用getProduct
+        //（由于Effect依赖了productList，若不加此判断，getProductList()方法将会不停调用,以至于productList数据重复紊乱）
+        if (productList.length === 0) {
+            getProductList()
+        }
+
+        //监听滚动
+        const handleScroll = () => {
+            const { innerHeight } = window    //显示屏height
+            let { scrollTop } = document.documentElement  //滚动出去的height
+            const { scrollHeight } = document.documentElement //文档高度
+
+            if (cutArrKey >= cutArr.length - 1) {
+                setIsLoading(false)
+                return;
+            }
+            //若触底了，向productList添加商品数据
+            if (innerHeight + scrollTop >= scrollHeight) {
+                setTimeout(() => {
+                    cutArrKey++
+                    let newArr = [...productList, ...cutArr[cutArrKey]]
+                    setProductList(newArr)
+                }, 1000)
+
+            }
+        }
+        window.addEventListener('scroll', handleScroll)
```

Comment:
Please throttle the scroll event handler, otherwise it will be called over 100 times while scrolling the page.